shared/src/api/interceptors.coffee

```diff
@@ -0,0 +1,134 @@
+_ = require 'lodash'
+minimatch = require 'minimatch'
+{Promise} = require 'es6-promise'
+
+makeLocalRequest = (requestConfig) ->
+  {url} = requestConfig
+
+  [uri, params] = url.split('?')
+  if requestConfig.method is 'GET'
+    url_parts = ["#{uri}.json", params]
+  else
+    url_parts = ["#{uri}/#{requestConfig.method}.json", params]
+    requestConfig.mockMethod = requestConfig.method
+    requestConfig.method = 'GET'
+
+  requestConfig.url = _.compact(url_parts).join('?')
+
+makeLocalResponse = (response) ->
+  payload = if response.config.data then JSON.parse(response.config.data) else {}
+  response.data = _.extend({}, response.data, payload)
+
+doesErrorMatch = (handledErrors, errorName) ->
+  _.indexOf(handledErrors, errorName) > -1 or
+    _.some(handledErrors, _.partial(minimatch, errorName))
+
+areAllErrorsHandled = (handledErrors, errors, errorNameProperty) ->
+  isErrorHandled = _.partial(doesErrorMatch, handledErrors)
+  _.every errors, _.flow(_.property(errorNameProperty), isErrorHandled)
+
```

Comment:
ah, this is meant to be a non-nested version of this

```
areAllErrorsHandled = (handledErrors, errors, errorNameProperty) ->
  _.every errors, (error) ->
    doesErrorMatch(handledErrors, _.property(errorNameProperty)(error))
```

but perhaps the nested version is easier to understand in this case.  actually, i think i would prefer this:

```
areAllErrorsHandled = (handledErrors, errors, errorNameProperty) ->
  _.every errors, (error) ->
    errorName = _.property(errorNameProperty)(error)
    doesErrorMatch(handledErrors, errorName)
```